### PR TITLE
Add controller + model listing OEM meta-packages

### PR DESCRIPTION
**Pull request by ogayot**
This is one step towards what we need for OEM. Natural next steps are:

* Making sure that the drivers screen does not list OEM meta-packages (metapackages are not drivers and should be installed unconditionally).
  * the `--no-oem` option could come to the rescue but currently we use the ubuntu-drivers command from the target rootfs. Old ISOs will not have it and will fail on installer refresh.
* Installing the OEM meta-packages. This is tricky because:
  * running `ubuntu-drivers install oem-somerville-tentacool-meta` or alike does not work - the package does not get installed
  * running `ubuntu-drivers install` will install third-party drivers even if the user decided not to do so.

### CODE REVIEW

File: subiquity/server/controllers/oem.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    model_name = "oem"
+
+    def __init__(self, app) -> None:
+        super().__init__(app)
+        # At this point, the source variant has not been selected but it only
+        # has an impact if we're listing drivers, not OEM metapackages.
+        self.ubuntu_drivers = get_ubuntu_drivers_interface(self.app)
+
+        self.task: Optional[asyncio.Task] = None
```
Comment (dbungert) [inline subiquity/server/controllers/oem.py:47]: nitpick: a more verbose name on self.task might be nice, where we have one task we tend to accumulate more.

**Review by dbungert — APPROVED** [review]

File: subiquity/server/controllers/oem.py (hunk 1 above)
Comment (ogayot) [reply subiquity/server/controllers/oem.py]: Updated, thanks!

**Review by dbungert — APPROVED** [review]

**Review by mwhudson — COMMENTED** [review]
Hi I realize I am late coming to this and sorry about that, but what is the API actually for here? Running list-oem when apt is configured --> great, telling the client about it --> why? Didn't I browbeat you into agreeing that we just install the oem packages without any client involvement?

**Comment by ogayot** [comment]
> Hi I realize I am late coming to this and sorry about that, but what is the API actually for here? Running list-oem when apt is configured --> great, telling the client about it --> why? Didn't I browbeat you into agreeing that we just install the oem packages without any client involvement?

Yes, we agreed on that part. The API endpoint is probably not useful for the client and I did not implement it at first. It felt like a natural way to check what's going on (mostly for us) but I don't mind reverting the last patch (which adds the GET endpoint) if needed. What do you think?

**Comment by mwhudson** [comment]
> > Hi I realize I am late coming to this and sorry about that, but what is the API actually for here? Running list-oem when apt is configured --> great, telling the client about it --> why? Didn't I browbeat you into agreeing that we just install the oem packages without any client involvement?
> 
> Yes, we agreed on that part. The API endpoint is probably not useful for the client and I did not implement it at first. It felt like a natural way to check what's going on (mostly for us) but I don't mind reverting the last patch (which adds the GET endpoint) if needed. What do you think?

Eh it's implemented already and it lets you write tests it may as well stay. Just wanted to check that's all :)

**Pull request merged**